pywhy_stats/kci.py

```diff
@@ -0,0 +1,196 @@
+from typing import Callable
+
+import numpy as np
+from numpy import shape, ndarray
+from numpy.linalg import eigvalsh
+from numpy.random import permutation
+from scipy import stats
+
+from pywhy_stats.custom_kernels import rbf_kernel
+from pywhy_stats.pvalue_result import PValueResult
+
+
+def ind(X: np.ndarray,
+        Y: np.ndarray,
+        kernel_X: Callable[[np.ndarray], np.ndarray] = rbf_kernel,
+        kernel_Y: Callable[[np.ndarray], np.ndarray] = rbf_kernel,
```

Comment:
We could support passing in a Callable, but also a `str`, which is of any possibility in sklearn's pairwise kernel metrics. This ensures that we have a wide-range of kernels for free. 

E.g. see: https://hyppo.neurodata.io/api/generated/hyppo.independence.hsic

Reply:
If a user want's to use a scikit kernel, simply pass it by:
```
from sklearn.metrics.pairwise import polynomial_kernel
ind(X, Y, kernel_X = polynomial_kernel)
```
thats all they need to do. And we avoid having a long list of possible strings.

Reply:
Another advantage is, the kernels can also be pre-configure via partials, which will not be possible if we just allow a string to select it.

Reply:
> If a user want's to use a scikit kernel, simply pass it by:
> 
> ```
> from sklearn.metrics.pairwise import polynomial_kernel
> ind(X, Y, kernel_X = polynomial_kernel)
> ```
> 
> thats all they need to do. And we avoid having a long list of possible strings.

I like that. This is a good example to include in the `Examples` section in the docstring.